JPetRawSignal, JPetEvent support

grey277 opened the pull request:
Add support for JPetRawSignal, JPetEvent, change data structure to singleton, move creating geometry to GeometryVirtualizator, some refactoring of GeometryVirtualization, remove load geometry button(in final version geometry will be loaded from parambank, now geometry is hardcoded in eventdisplay), not working multiple diagrams( Divide on diagram canvas not creating pads )

grey277 commented:
I updated code, now there is working checkbox for option to save markers and LORs between events(default no), also top view is finished, markers are drawn directly by x,y coords, I scaled canvas by rescale function. diagrams working correctly,  "start virt" button now displays next 100 events with 10ms sleep, I think after code review it could be accepted as 0.2v

For not working things:
- on 2d view markers are drawn still in middle of scin and they aren't clear after next event
- there could be extra controls for automatic visualization, for number of events and time sleep
- update info panel, for now there is only activated scin id and layer id
- move mapping geometry to DataProcessor constructor and display message box when loading mapping geometry

wkrzemien commented on CMakeLists.txt:
```diff
@@ -48,21 +48,21 @@ link_directories(${LIBRARY_PARENT})
-  CreateGeometry ALL
-  DEPENDS JPET_geom.root
-)
+#add_custom_command(
```
This can be removed?

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -136,40 +135,132 @@ DataProcessor::getActiveScintillators(const JPetRawSignal &rawSignal)
+void DataProcessor::getActiveScintillators(const JPetHit &hitSignal)
+{
+
+  //std::cout << "x: " << hitSignal.getPosX() << " y: " << hitSignal.getPosY()
```
remove ?

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -136,40 +135,132 @@ DataProcessor::getActiveScintillators(const JPetRawSignal &rawSignal)
+  ScintillatorsInLayers selection;
+  for(JPetHit hit : event.getHits())
+  {
+    //std::cout << "x: " << hit.getPosX() << " y: " << hit.getPosY()
```
remove

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+      << "\n";
+  JPetParamManager fparamManagerInstance(
+      new JPetParamGetterAscii("large_barrel.json"));
+  fparamManagerInstance.fillParameterBank(44);
```
at least extract 44 as a local variable and add some comment. Is this a run number?

wkrzemien replied:
the same with large_barrel.json name of the file.

The best would be to hold those in some json file and load it in the constructor

wkrzemien commented:
Please add somewhere (maybe README) explicit info which j-pet-framework version is used (probably commit + tag if relevant)

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+    getHitsPosition(getCurrentEvent< JPetEvent >());
+    break;
+  default:
+    break;
```
Write something if the type is unknown?

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
-    if (selection.find(pos.layer) != selection.end()) {
+    if (selection.find(pos.layer) != selection.end())
+    {
       if (std::find(selection[pos.layer].begin(), selection[pos.layer].end(),
```
If you don't mind. I would extract the code inside it to a inline function. It will be easier to read
e.g. addSlotIfNotPresent or something similar.

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
-    if (selection.find(pos.layer) != selection.end()) {
+    if (selection.find(pos.layer) != selection.end())
+    {
       if (std::find(selection[pos.layer].begin(), selection[pos.layer].end(),
```
as above -> one can reuse the same function?

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+  for (JPetHit hit : event.getHits())
+  {
+    StripPos pos = fMapper->getStripPos(hit.getBarrelSlot());
+    if (selection.find(pos.layer) != selection.end())
```
and here as well

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+  for (auto it = data.begin(); it != data.end(); it++)
+  {
+    diff[i] = startPos - it->second.second;
+    if (diff[i] > 0)
```
Add at least a comment to explain this condition diff[i] >0 I think it is not obvious

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
-  if (fCurrentFileType == FileTypes::fRawSignal)
-    data = getDataForDiagram(
-        dynamic_cast<JPetRawSignal &>(fReader.getCurrentEvent()));
+DiagramDataMap DataProcessor::getDataForDiagram(const JPetRawSignal &rawSignal,
```
This function requires some explanation, what you exactly do inside

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+DiagramDataMap DataProcessor::getDataForDiagram(const JPetRawSignal &rawSignal,
+                                                bool)
+{
+  DiagramDataMap r;
```
one letter variable should be used only as counter in loops

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+                                            .getPoints(JPetSigCh::Leading)[0]
+                                            .getPM()
+                                            .getBarrelSlot());
+    double r =
```
I understand it, but add some comment here, what is going on

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+
+void DataProcessor::getHitsPosition(const JPetEvent &event)
+{
+  HitPositions hits;
```
hits ->hitsPos ?

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+
+bool DataProcessor::openFile(const char *filename)
+{
+  static std::map< std::string, int > compareMap;
```
why  is it static?

grey277 replied:
If someone will open multiple files on one opening of application, it wont be needed to create compareMap on each call of openFile, i can remove it, I don't thing it give a lot of performance/memory boost.

wkrzemien commented on src/DataProcessor.cpp:
```diff
@@ -18,217 +18,366 @@
+    compareMap["JPetHit"] = FileTypes::fHit;
+    compareMap["JPetEvent"] = FileTypes::fEvent;
   }
   bool r = fReader.openFileAndLoadData(filename);
```
avoid one-letter variables

wkrzemien commented on src/DataProcessor.h:
```diff
@@ -16,65 +16,138 @@
+                                 JPetPM::Side, size_t, size_t > >
+    DiagramDataMap; // threshold number, thresholdValue, time, EdgeType, Side,
+                    // layer, scin
+// typedef std::map<int, std::tuple<float, float, JPetSigCh::EdgeType>>
```
remove commented code

wkrzemien commented on src/DataProcessor.h:
```diff
@@ -16,65 +16,138 @@
+  void setCurrentFileType(FileTypes type) { fCurrentFileType = type; }
+  void setHits(HitPositions hits) { fHits = hits; }
+
+  inline FileTypes getCurrentFileType() { return fCurrentFileType; }
```
since you are returning by value, probably that can be functionName() const ; ?

wkrzemien commented on src/DataProcessor.h:
```diff
@@ -16,65 +16,138 @@
+  DataProcessor &operator=(const DataProcessor &) = delete;
 
-  std::string getDataInfo(); // change when imp new mapper
+  void getActiveScintillators(const JPetTimeWindow &tWindow);
```
the name of functions getActiveScintillators are a bit misleading, cause you would typically expect that:
1.they return activ scitns list 2.they fill some list given explicitly as argument.
Maybe change the name or you could really return the list/vector/ here (which can be also easier to test)

wkrzemien commented on src/EventDisplay.cpp:
```diff
@@ -76,128 +80,166 @@ void EventDisplay::run()
   baseFrame->Resize(baseFrame->GetDefaultSize());
 
-  fMainWindow->SetWindowName("Single Strip Event Display ver 0.1");
+  fMainWindow->SetWindowName("Event Display ver 0.2");
```
J-PET Event Display ver 0.2

wkrzemien commented on src/EventDisplay.cpp:
```diff
@@ -76,128 +80,166 @@ void EventDisplay::run()
+                        kLHintsExpandX | kLHintsExpandY, 2, 2, 2, 2);
 
-  AddButton(frame1_1_2, "Read Geometry", "handleMenu(=0)");
+  // AddButton(frame1_1_2, "Read Geometry", "handleMenu(=0)");
```
remove

wkrzemien commented on src/EventDisplay.cpp:
```diff
@@ -262,60 +309,53 @@ void EventDisplay::handleMenu(Int_t id)
-      CloseWindow();
-    }
-    break;
+  /*case E_OpenGeometry:
```
to remove

wkrzemien commented on src/EventDisplay.cpp:
```diff
@@ -335,19 +376,36 @@ void EventDisplay::showData()
-  visualizator->drawStrips(selection);
-  visualizator->drawDiagram(dataProcessor->getDataForDiagram());
+  startVirtualizationLoop(10);
+  // std::thread t(&EventDisplay::startVirtualizationLoop, this, 1);
```
to remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
 
-  GeometryVisualizator::~GeometryVisualizator() { }
+  draw2dGeometry();
+  // setAllStripsUnvisible();
```
to remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+  assert(gPad);
+  TView *view = gPad->GetView();
+  assert(view);
+  view->ZoomView(gPad, 1.75);
```
for the sake of future changes add a comment what SetView(0,90,0, irep) does

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+  const int leftMargin = 20;
+  const int rightMargin = 20;
+  const int canvasScale = 900;
+  // int startX = canvasScale - leftMargin;
```
to remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    for (unsigned int j = 0; j < numberOfScintillatorsInCurrentLayer; j++)
+    {
+      fUnRolledViewScintillators[i][j] =
+          new TBox(leftMargin + (layerWidth * i),
```
Ok, here definitely write with words what are you doing

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    fUnRolledViewMarker.clear();
+  }
+
+  static const double centerOfScintillator = 25;
```
centerOfScintillator -> kCenterOfScintillator 
why is this static?

grey277 replied:
That each call of setMarker2d don't need to initialize centerOfScintillator variable, it can be removed if it is unwanted, also centerOfScintillator should be calculated as 1/2 of scintillator lenght, not hardcoded, I will change it.

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+          strip < fUnRolledViewScintillators[layer].size())
       {
-        allScintilatorsCanv[i][j].image->SetFillColor(1);
+        fUnRolledViewScintillators[layer][strip]->SetFillColor(2);
```
color type to variable?

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    if(layer == JPetGeomMapping::kBadLayerNumber)
+    {
+      std::cout << "/* Bad layer number in setVisibility, returning...*/" << "\n";
+      //Error("Bad layer number in setVisibility, returning...");
```
to remove ?

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+  const double axisPos = 59;
+  fCanvasTopView->cd();
+  fCanvasTopView->Range(-canvasRange, -canvasRange, canvasRange, canvasRange);
+  TEllipse *thirdLayerCircle = new TEllipse(0, 0, 57.5, 57.5);
```
extract those magic numbers at least to local variables

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
 
-  void GeometryVisualizator::setVisibility(const std::map<int, std::vector<int> >& selection)
+  TGaxis *axisY =
+      new TGaxis(axisPos, -57.5, axisPos, 57.5, -57.5, 57.5, 50510, "");
```
the same here, extract magic numbers

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    std::vector< float > leadingY;
+    std::vector< float > trailingX;
+    std::vector< float > trailingY;
+    const float changePsToNs = 0.001;
```
kPsToNs

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    mg->GetYaxis()->SetTickLength(0);
+    mg->GetYaxis()->SetRangeUser(0, 5);
+    mg->GetXaxis()->SetLimits(0, 50);
+    // mg->GetXaxis()->SetLimits(mg->GetXaxis()->GetXmin() - 10 * 1000,
```
remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    mg->GetXaxis()->SetTitle("Time (ns)");
+    mg->GetYaxis()->SetTitle("Threshold Number");
+
+    for (int k = 0; k < 4; k++)
```
4 can be also extracted

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+float GeometryVisualizator::changeSignalNumber(
+    int signalNumber) // change this...
+{
+  // std::cout << "signalNumber: " << signalNumber << "\n";
```
remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
-    name.append(CommonTools::intToString(layer) + std::string("_1"));
-    return name;
-  }
+float GeometryVisualizator::changeSignalNumber(
```
I dont understand this function

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+  newaxis->Draw();
+}
+
+void GeometryVisualizator::createGeometry(
```
Add some function description

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
+    for (int j = 0; j < layerStats[i].first; j++)
+    {
+      TGeoVolume *scin = gGeoManager->MakeTube("scin", medium, 0, 0.7, 50);
+      // scin->SetLineWidth(5);
```
remove

wkrzemien commented on src/GeometryVisualizator.cpp:
```diff
@@ -25,287 +25,586 @@
   }
+  fGeoManager->CloseGeometry();
+  fGeoManager->SetVisLevel(4);
+  // fGeoManager->Export("tt.root");
```
remove

wkrzemien commented on src/GeometryVisualizator.h:
```diff
@@ -16,71 +16,146 @@
+
+  void draw2dGeometry2();
+
+  enum ColorTable
```
since you have it defined you can use it in the code -> sometimes you have been using raw numbers

wkrzemien commented on src/GeometryVisualizator.h:
```diff
@@ -16,71 +16,146 @@
+  ColorTable layersColors[3]{kGray1, kGray2, kGray3};
+#endif
+
+  /*struct ScintillatorCanv
```
to remove

grey277 commented:
For now, i extracted variables defining DataProcessor/createGeometry to EventDisplay and they can be passed when creating EventDisplay, when UserOptions will be accepted I will change it for new version of options. Also I updated README file.
